src/AstClause.h

```diff
@@ -183,45 +182,6 @@ class AstClause : public AstNode {
         return toPtrVector(bodyLiterals);
     }
 
-    /**
-     * Re-orders atoms to be in the given order.
-     * Remaining body literals remain in the same order.
-     **/
-    // TODO (b-scholz): remove this method
-    void reorderAtoms(const std::vector<unsigned int>& newOrder) {
-        std::vector<unsigned int> atomPositions;
```

Comment:
I am concerned with the latest changes whether the reordering is still working. The reason is that the reordering numbers were related to atoms in a body only.  However, in the new system, we may refer to atoms in schedules differently, ie., constraints/negations are counted as well. Is this correct?

Reply:
I took this into account when writing the replacement atom-reordering function; the function is semantically equivalent to filtering out all non-atoms, reordering the atoms based on the new order, and then putting back the non-atoms where they were before.

So, something like ‘a(x) :- b(x), !c(x), d(x)’ with new order ‘[1, 0]’ should result in ‘a(x) :- d(x), !c(x), b(x).’

 Good point, though - I will add a test to check for this.